network/ipv6Transport.js

```diff
@@ -0,0 +1,263 @@
+                    return addresses[0].address;
+            }
+            return undefined;
+        }
+
+        /**
+         * Check IP address belongs to local interfaces
+         * @param {string} ip 
+         */
+        _checkLocalInterfacesIp(ip) {
+            const interfaces = os.networkInterfaces();
+            for (let interfaceName in interfaces) {
+                if ((interfaces[interfaceName].filter(iface => {
+                    return iface.address === ip;
+                })).length > 0) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /**
+         *
+         * @param name
+         * @return {Promise<*|string[]>}
+         */
+        static async resolveName(name) {
+            try {
+                const dnsResolveDelegate = util.promisify(dns.resolve);
+                return await dnsResolveDelegate(name, 'ANY');
+            } catch (err) {
+                debug(`dnsName: ${name}; error: ${err}`);
+                return null;
+            }
+        }
+
+        _portMapping() {
+            if (!this._upnpClient) {
+                this._upnpClient = natUpnp.createClient();
+            }
+            this._upnpClient.portMapping({
+                public: this._port,
+                private: this._port,
+                ttl: 0
+            }, function (err) {
+                if (err) {debug(`portMapping error`, err);}
+            });
+        }
+
+        /**
+         * For tests
+         */
+        portUnmapping() {
+            if (this._upnpClient) {
+                this._upnpClient.portUnmapping({
+                    public: this._port
+                });
+            }
+        }
+
+        /**
+         * @param {String} address - IP address
+         * @return {Promise<Ipv6Connection>} new connection
+         */
+        connect(address, port) {
+            return new Promise((resolve, reject) => {
+                const socket = net.createConnection(port, address,
+                    async (err) => {
+                        if (err) return reject(err);
+                        resolve(new Ipv6Connection({socket, timeout: this._timeout}));
+                    }
+                );
+                socket.on('error', err => reject(err));
+            });
+        }
+
+        /**
+         * uPnP port mapping, set result addresses
+         */
+        async _getMappingAddress() {
+            let mappings = await this._getPortMappings();
+            if (!mappings || mappings.length === 0) {
+                this._portMapping();
+                mappings = await this._getPortMappings();
+            }
+            if (mappings && mappings.length > 0) {
+                return {privateAddress: mappings[0].private.host};
+            }
+        }
+
+        _setAddresses({privateAddress, publicAddress}) {
+            this._privateAddress = !this.privateAddress ? privateAddress || undefined : this.privateAddress;
```

Comment:
Зачем столь дикие конструкции?

```suggestion
            this._privateAddress = this.privateAddress || privateAddress;
```